src/ScType.ts

```diff
@@ -38,79 +54,116 @@ export class ScType {
     return (this._value & sc_type_constancy_mask) != 0;
   }
 
+  public isConst(): boolean {
+    return (this._value & sc_type_const) != 0;
+  }
+
+  public isVar(): boolean {
+    return (this._value & sc_type_var) != 0;
+  }
+
   public hasDirection(): boolean {
-    return (this._value & sc_type_uedge_common) == 0;
+    return (this._value & sc_type_common_edge) == 0;
   }
 
   public isNode(): boolean {
     return (this._value & sc_type_node) != 0;
   }
 
+  public isLink(): boolean {
+    return (this._value & sc_type_node_link) == sc_type_node_link;
+  }
+
+  public isConnector(): boolean {
+    return (this._value & sc_type_connector) != 0;
+  }
```

Comment:
why are some methods comparing result with 0 and some are comparing result with type? also a lot of duplicated code, can be in some method that takes type and returns boolean